### PR TITLE
Add menu to editor

### PR DESCRIPTION
**Pull request by jannessm**

### CODE REVIEW

File: src/app/components/preview/preview.component.ts [hunk 1]
```diff
@@ -7,35 +7,65 @@ import { HtmlFactoryService } from '../../services/html-factory.service';
...
-  private zoom:number = 1;
+  private zoom = 1;
 
-  constructor(private htmlFactory:HtmlFactoryService, private renderer: Renderer2) {}
+  @HostListener('window:keyup', ['$event', '$event.keyCode'])
+  scroll(e, code) {
+    if (this.performMode) {
+      e.preventDefault();
+      switch (code) {
+        case 13:
```
Comment (nilsroesel) [inline src/app/components/preview/preview.component.ts]: why empty cases?

File: src/app/components/preview/preview.component.ts [hunk 2]
```diff
@@ -7,35 +7,65 @@ import { HtmlFactoryService } from '../../services/html-factory.service';
...
+      }
+    }
+  }
+
+  constructor(private htmlFactory: HtmlFactoryService, private renderer: Renderer2) {}
 
-  ngOnInit(){
-    if(!this.song)
+  ngOnInit() {
+    if (!this.song) {
```
Comment (nilsroesel) [inline src/app/components/preview/preview.component.ts]: shorter
```typescript
this.song = this.song || new Song();
```

File: src/app/components/songsheet-textarea/songsheet-textarea.component.ts [hunk 3]
```diff
@@ -126,4 +134,11 @@ export class SongsheetTextareaComponent implements OnInit, OnChanges {
     return i;
   }
 
+  /*public addResolveSymbol() {
```
Comment (nilsroesel) [inline src/app/components/songsheet-textarea/songsheet-textarea.component.ts]: ???

File: src/app/views/editor/editor.component.ts [hunk 4]
```diff
@@ -35,8 +41,19 @@ export class EditorComponent implements OnInit {
     this.songIn = song;
   }
 
-  songOut(song){
-    this.songIn = song;
+  save() {
```
Comment (nilsroesel) [inline src/app/views/editor/editor.component.ts]: empty function

File: src/app/views/editor/editor.component.ts [hunk 5]
```diff
@@ -35,8 +41,19 @@ export class EditorComponent implements OnInit {
...
+
+  }
+
+  performMode() {
+    this.router.navigateByUrl('perform/' + this.songId);
   }
 
+  transposeUp() {}
```
Comment (nilsroesel) [inline src/app/views/editor/editor.component.ts]: empty function

File: src/app/views/editor/editor.component.ts [hunk 6]
```diff
@@ -35,8 +41,19 @@ export class EditorComponent implements OnInit {
...
+
+  transposeDown() {}
```
Comment (nilsroesel) [inline src/app/views/editor/editor.component.ts]: empty function

File: src/app/views/editor/editor.component.ts [hunk 7]
```diff
@@ -35,8 +41,19 @@ export class EditorComponent implements OnInit {
...
+
+  addResolveSymbol() {
```
Comment (nilsroesel) [inline src/app/views/editor/editor.component.ts]: empty function.

File: src/app/components/preview/preview.component.ts (hunk 1 above)
Comment (jannessm) [reply src/app/components/preview/preview.component.ts]: there are not empty. all cases results in the same function...

File: src/app/components/songsheet-textarea/songsheet-textarea.component.ts (hunk 3 above)
Comment (jannessm) [reply src/app/components/songsheet-textarea/songsheet-textarea.component.ts]: will be added another time. its a placeholder as a reminder

File: src/app/views/editor/editor.component.ts (hunk 4 above)
Comment (jannessm) [reply src/app/views/editor/editor.component.ts]: placeholder...

File: src/app/views/editor/editor.component.ts (hunk 5 above)
Comment (jannessm) [reply src/app/views/editor/editor.component.ts]: placeholder...

File: src/app/views/editor/editor.component.ts (hunk 6 above)
Comment (jannessm) [reply src/app/views/editor/editor.component.ts]: placeholder...

File: src/app/views/editor/editor.component.ts (hunk 7 above)
Comment (jannessm) [reply src/app/views/editor/editor.component.ts]: placeholder...

**Pull request merged**